rewrite-gradle/src/main/java/org/openrewrite/gradle/UpgradeDependencyVersion.java

```diff
@@ -116,189 +121,259 @@ public Validated<Object> validate() {
-                        if (!maybeGp.isPresent()) {
-                            return cu;
-                        }
+    public static class DependencyVersionState {
+        GradleProject gradleProject;
+        List<Expression> dependencyExpressions = new ArrayList<>();
+    }
 
-                        cu = (JavaSourceFile) new UpdateVariable(variableNames, maybeGp.get()).visitNonNull(cu, ctx);
-                    }
-                    Map<GroupArtifactVersion, Set<String>> versionUpdates = getCursor().getMessage(NEW_VERSION_KEY);
-                    if (versionUpdates != null) {
-                        Optional<GradleProject> maybeGp = cu.getMarkers()
-                                .findFirst(GradleProject.class);
-                        if (!maybeGp.isPresent()) {
-                            return cu;
-                        }
-                        GradleProject newGp = maybeGp.get();
-                        for (Map.Entry<GroupArtifactVersion, Set<String>> gavToConfigurations : versionUpdates.entrySet()) {
-                            newGp = replaceVersion(newGp, ctx, gavToConfigurations.getKey(), gavToConfigurations.getValue());
-                        }
-                        cu = cu.withMarkers(cu.getMarkers().removeByType(GradleProject.class).add(newGp));
-                    }
-                    return cu;
-                }
-                return tree;
-            }
+    @Override
+    public DependencyVersionState getInitialValue(ExecutionContext ctx) {
+        return new DependencyVersionState();
+    }
 
+    @Override
+    public TreeVisitor<?, ExecutionContext> getScanner(DependencyVersionState acc) {
+        MethodMatcher dependencyDsl = new MethodMatcher("DependencyHandlerSpec *(..)");
+        return new GroovyVisitor<ExecutionContext>() {
             @Override
             public J visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
                 J.MethodInvocation m = (J.MethodInvocation) super.visitMethodInvocation(method, ctx);
                 if (dependencyDsl.matches(m)) {
                     List<Expression> depArgs = m.getArguments();
-                    if (depArgs.get(0) instanceof J.Literal || depArgs.get(0) instanceof G.GString || depArgs.get(0) instanceof G.MapEntry) {
-                        m = updateDependency(m, ctx);
-                    } else if (depArgs.get(0) instanceof J.MethodInvocation &&
-                            (((J.MethodInvocation) depArgs.get(0)).getSimpleName().equals("platform") ||
-                                    ((J.MethodInvocation) depArgs.get(0)).getSimpleName().equals("enforcedPlatform"))) {
-                        m = m.withArguments(ListUtils.mapFirst(depArgs, platform -> updateDependency((J.MethodInvocation) platform, ctx)));
+                    if (depArgs.get(0) instanceof J.Literal ||
+                        depArgs.get(0) instanceof G.GString ||
+                        depArgs.get(0) instanceof G.MapEntry) {
```

Comment:
Perhaps I'm missing something, but further down we seem to only process `G.GString`; could we then not add the others here?
```suggestion
                    if (depArgs.get(0) instanceof G.GString) {
```

Reply:
If so we can also change the field type on `DependencyVersionState`, and remove the type cast in `visitEntry` below.

Reply:
So I agree partially. Both the `G.GString` and `G.MapEntry` could have an identifier for the version value. Only the `J.Literal` would be valid to skip.

Reply:
Should we then add handling for G.MapEntry below as well? Or is that out of scope for now?

Reply:
You reminded me about missing that, I've added a comment below to that effect. :)

Reply:
Great; then I suppose the proper suggestion here is to use
```suggestion
                    if (depArgs.get(0) instanceof G.GString ||
                        depArgs.get(0) instanceof G.MapEntry) {
```